From: Alexander Mikhaylenko <alexander.mikhaylenko@puri.sm>
Date: Tue, 22 Sep 2020 15:19:27 +0500
Subject: librem5: adblock: Use GTimeVal again instead of GDateTime

We have too old glib, so this will have to do for now

Reverts https://gitlab.gnome.org/GNOME/epiphany/commit/8ac1e62b72b47e98217377c9737557f4900e338d
---
 embed/ephy-filters-manager.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/embed/ephy-filters-manager.c b/embed/ephy-filters-manager.c
index 0ca319e..b2c465f 100644
--- a/embed/ephy-filters-manager.c
+++ b/embed/ephy-filters-manager.c
@@ -401,7 +401,7 @@ filter_info_needs_updating_from_source (const FilterInfo *self)
    * than the last update time saved for it.
    */
   if (self->local) {
-    g_autoptr (GDateTime) modification_time = NULL;
+    GTimeVal modification_time = { .tv_sec = 0 };
     g_autoptr (GError) error = NULL;
     g_autoptr (GFile) source_file = g_file_new_for_uri (self->source_uri);
     g_autoptr (GFileInfo) info = g_file_query_info (source_file,
@@ -414,8 +414,8 @@ filter_info_needs_updating_from_source (const FilterInfo *self)
       return TRUE;
     }
 
-    modification_time = g_file_info_get_modification_date_time (info);
-    return g_date_time_to_unix (modification_time) > self->last_update;
+    g_file_info_get_modification_time (info, &modification_time);
+    return (modification_time.tv_sec > 0) && (modification_time.tv_sec > self->last_update);
   }
 
   /*
